### ISSUE TITLE
Rule proposal: Style/ElseAlignment but for blocks

### ISSUE THREAD

**Issue by ptarjan** [rubocop-hq/rubocop#2883]
```
$ cat /tmp/a.rb
foo =
  if bar
      1
    else
      baz
    end
puts foo

# Please complain about this block the same way as the first
foo =
  begin
      1
    rescue
      baz
    end
puts foo

$ rubocop /tmp/a.rb -D
Inspecting 1 file
W

Offenses:

/tmp/a.rb:3:3: C: Style/IndentationWidth: Use 2 (not 4) spaces for indentation.
      1
  ^^^^
/tmp/a.rb:4:5: C: Style/ElseAlignment: Align else with if.
    else
    ^^^^
/tmp/a.rb:6:5: W: Lint/EndAlignment: end at 6, 4 is not aligned with if at 2, 2.
    end
    ^^^

1 file inspected, 3 offenses detected
```

**Comment by jonas054** [comment]
Thanks for spotting this problem. Here's what's wrong IMO:

The `end` keyword is not aligned with `begin` and we totally miss that. Calling `ruby -wc` on the code gives us a warning for it so this problem should be reported by a `Lint` cop, and I think a new cop would be in order, e.g. `Lint/BeginEndAlignment`.

The other lines inside `begin`/`end` *are* checked today. Their indentations are checked relative to the `end` keyword.

**Comment by bbatsov** [comment]
There hasn't been much activity on this ticket and our Core Team is spread too thin on so many tasks, so I'll just close it for the sake of having a cleaner lists of tasks to focus on. We'd gladly review a PR, but it's unlikely that'd we're going to tackle this ourselves in the foreseeable future.
